content/_includes/post.macros.njk

```diff
@@ -245,15 +247,45 @@ params:
     default: true
     note: Optionally remove event links for content in linked-cards
 #}
-{% macro upcoming(
+{% macro eventpreview(
   events,
   links=true
 ) %}
-  {%- set events = events | getFuture if events else none -%}
+  {%- set events = events -%}
   {%- set events = events | sort(attribute='date') -%}
-  {%- if events | length > 0 -%}
+  {%- set pastevents = events -%}
+  {%- set futureevents = events | getFuture  -%}
```

Comment:
It seems like these two lists are mostly duplicated. We could maybe simplify by doing something like:

```njk
{%- set futureevents = events | getFuture -%}
{%- set isfuture = futureevents | length > 0 -%}
{%- set showevents = futureevents if isfuture else events | reverse -%}

  {%- if showevents | length > 0 -%}
    <h4>{{ 'Upcoming' if isfuture else 'Past' }}</h4>
    <ul class="next-event">
    {%- for next in showevents -%}
      etc...
```

That first sorts out what event list to show, and then only runs the output loop once.